### ISSUE TITLE
Documentation for keymap_common?

### ISSUE THREAD

**Issue by aaronz8** [tmk/tmk_keyboard#186]
I've been following the tutorial at http://deskthority.net/workshop-f7/how-to-build-your-very-own-keyboard-firmware-t7177.html to try to use tmk with FaceW PCB. However, I am stuck at http://deskthority.net/workshop-f7/how-to-build-your-very-own-keyboard-firmware-t7177.html#p141386, where it skips over the keymap_common.h configuration. I tried looking for other examples, such as 
https://github.com/tmk/tmk_keyboard/blob/master/keyboard/nerd/keymap_common.h, but I'm still confused where these numbers come from.

Is there documentation for this anywhere?

image of the FaceW row,column connections for each key:
![screen shot 2015-02-11 at 1 00 34 am](https://cloud.githubusercontent.com/assets/940275/6144206/6d097cdc-b189-11e4-9f61-e18bacb817fe.png)

**Comment by aaronz8** [comment]
I've figured it out. I'll do a write up when I've tested everything.

**Comment by Leimi** [comment]
Hey,

Kinda late to the party but did you end up getting something working/writing down anything on it? I'd love to get my FaceW PCB working with TMK but I'm terrible at all this.

**Comment by aaronz8** [comment]
Hi @Leimi,
Sorry for the delayed response. Unfortunately I soon discovered that the FaceW PCB is based around the Atmega32a, which does not have integrated USB and thus not supported by the tmk firmware. The only bootloader I could find supporting the atmega32a was [bootloadhid](https://www.obdev.at/products/vusb/bootloadhid.html), along with its virtual usb implementation, [V-USB] (https://www.obdev.at/products/vusb/index.html). I looked into using V-USB with tmk but gave up once I found out it requires libusb and libusb doesn't work well with HID on Mac (http://www.libusb.org/ticket/89).
